### ISSUE TITLE
Rename `JobInfo`

### ISSUE THREAD

**Issue by HereThereBeDragons** [cvmfs/cvmfs#3217]
S3 Upload and Downloadmanager use that same name `JobInfo` for different structs.

To aid with readability it would be good to rename them:
For Downloadmanager: `DownloadInfo` or `DownloadJob`
For S3 Upload: `UploadInfo` or `S3UploadInfo`, same here `Info` could be replaced with `Job`

**Comment by HereThereBeDragons** [comment]
instead of `info` or `job` better use `context`

**Comment by HereThereBeDragons** [comment]
Part of #3095

**Comment by Agamya-Samuel** [comment]
hi @vvolkl @HereThereBeDragons 
Is this issue still being worked upon ?

**Comment by vvolkl** [comment]
Yes, but not with any priority at the moment. This issue is also part of a larger refactoring. While I may accept patches here, it will probably need a lot of discussion.
